solr/core/src/java/org/apache/solr/cloud/CreateAliasCmd.java

```diff
@@ -68,34 +249,100 @@ public void call(ClusterState state, ZkNodeProps message, NamedList results)
+                  "(Hint: 'NOW' used without rounding always has this problem)" );
+        }
+        return date.toInstant();
+      } catch (SolrException e) {
+        throw new SolrException(BAD_REQUEST,
+            "Start Time for the first collection must be a timestamp of the format yyyy-MM-dd_HH_mm_ss, " +
+                "or a valid date math expression not containing specific milliseconds", e);
+      }
+    }
+    return Instant.from(startTime);
+  }
+
+  private TemporalAccessor attemptTimeStampParsing(String start, ZoneId zone) {
+    try {
+      DATE_TIME_FORMATTER.withZone(zone);
+      return DATE_TIME_FORMATTER.parse(start);
+    } catch (DateTimeParseException e) {
+      return null;
+    }
+  }
+
+  private boolean anyRoutingParams(ZkNodeProps message) {
+
+    return message.containsKey(ROUTING_FIELD) || message.containsKey(ROUTING_TYPE) || message.containsKey(START)
+        || message.containsKey(ROUTING_INCREMENT) || message.containsKey(TZ);
+  }
+
   private void validateAllCollectionsExistAndNoDups(List<String> collectionList, ZkStateReader zkStateReader) {
     final String collectionStr = StrUtils.join(collectionList, ',');
 
     if (new HashSet<>(collectionList).size() != collectionList.size()) {
-      throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+      throw new SolrException(BAD_REQUEST,
           String.format(Locale.ROOT,  "Can't create collection alias for collections='%s', since it contains duplicates", collectionStr));
     }
     ClusterState clusterState = zkStateReader.getClusterState();
     Set<String> aliasNames = zkStateReader.getAliases().getCollectionAliasListMap().keySet();
     for (String collection : collectionList) {
       if (clusterState.getCollectionOrNull(collection) == null && !aliasNames.contains(collection)) {
-        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
+        throw new SolrException(BAD_REQUEST,
             String.format(Locale.ROOT,  "Can't create collection alias for collections='%s', '%s' is not an existing collection or alias", collectionStr, collection));
       }
     }
   }
-  
+
   /**
    * The v2 API directs that the 'collections' parameter be provided as a JSON array (e.g. ["a", "b"]).  We also
    * maintain support for the legacy format, a comma-separated list (e.g. a,b).
    */
   @SuppressWarnings("unchecked")
   private List<String> parseCollectionsParameter(Object colls) {
-    if (colls == null) throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, "missing collections param");
+    if (colls == null) throw new SolrException(BAD_REQUEST, "missing collections param");
     if (colls instanceof List) return (List<String>) colls;
     return StrUtils.splitSmart(colls.toString(), ",", true).stream()
         .map(String::trim)
         .collect(Collectors.toList());
   }
 
+  private ZkNodeProps selectByPrefix(String prefix, ZkNodeProps source) {
```

Comment:
I think the use of the single element array as a holder should be avoided -- I mean I've done it in some circumstances but here you're only doing it to use Java 8 streams, which IMO isn't a good reason for that hack. Either use a standard loop construction, or use the version of Stream.collect that takes the supplier & accumulator to take your Java 8 kung-foo to the next level ;-)  'course most people reading code using such esoteric Java 8 stream features will need to read the docs to figure out what the heck is going on.  Trade-offs...

Reply:
Never really considered it a hack myself... it's just the standard way to work around the final variable restriction (now relaxed to "effectively final" of course) on anonymous inner classes. Lambdas (in java) are mostly just anon inner classes in disguise... but don't really care, changed to for loop in update.